### PR TITLE
Added a document on dotnet custom metrics

### PR DESCRIPTION
**Pull request by nico-shishkin**
# What changed

https://deploy-preview-1270--logz-docs.netlify.app/shipping/prometheus-sources/dotnet-custom.html

----

## Pages to review

- LinkToPage1
- LinkToPage2

## Remaining work

- [ ] Technical review
- [ ] Copy Review
- [ ] Redirects: 

## Post launch

To be completed by the docs team upon merge:

- [ ] Teams to update with the new information:
- [ ] Replace original content on the support portal with 'this page has been moved to ...' - paste the URL
- [ ] Update these log shipping pages in the app: - paste the URL

### CODE REVIEW

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+
+To create MetricsBuilder, copy and paste the following code into the function of the code that you need to export metris from:
+
+```java
+var metrics = new MetricsBuilder()
+                .Report.ToLogzioHttp("[[<<LISTENER-HOST>>]]", "[[<<METRICS-SHIPPING-TOKEN>>]]")
+                .Build();
+```
+
+{% include log-shipping/listener-var.html %}
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:76]: Please add to the listener examples the 'https://' or 'http://' and the port. If they use 'https://' the port is ':8053' and if they use 'http://' the port is ':8052'

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+Once the metrics are in Logz.io, you can query requited metrics using labels. Each metric has the following labels:
+
+| App Metrics parameter name | Description | Logz.io parameter name |
+|---|---|---|
+| Name | The name of the metric. Required for each metric. | Metric name if not stated otherwise |
+| MeasurementUnit | The unit you use to measure. By default it is `None`. | `unit` |
+| Context | The context which the metric belong to. By default it is `Application`.	 | `context` |
+| Tags | Pairs of key and value of the metric. It is not required to have tags for a metric.| Tags keys |
+
+Some of the metrics have custom labels as described below.
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:164]: You can write here that the clients should replace [[your_metric_name]] with their metric name.

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+
+Alternatively, If you prefer to install the library manually, download the latest version from NuGet Gallery
+
+
+##### Create MetricsBuilder
+
+To create MetricsBuilder, copy and paste the following code into the function of the code that you need to export metris from:
+
+```java
+
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:279]: You can remove the new line here :)

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+You can configure MetricsBuilder to use ToLogzioHttp exporter, which allows you to exprt metrics via HTTP using additional export settings. To enable this exporter, add the following code block to define the MetricsBuilder:
+
+```java
+var metrics = new MetricsBuilder()
+                .Report.ToLogzioHttp(options =>
+                {
+                    options.Logzio.EndpointUri = new Uri("[[logzio_endpoint]]");
+                    options.Logzio.Token = "[[metrics_token]]";
+                    options.FlushInterval = TimeSpan.FromSeconds(15);
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:482]: Please change [[logzio_endpoint]] and [[metrics_token]] to be like the variables in the explanation (<<LISTENER-HOST>>)

**Review by ShiranAvidov — CHANGES_REQUESTED** [review]
Amazing work. It looks really good. I commented about couple of things that I believe should be changed, please take a look at them :)

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+```csharp
+var counter = new CounterOptions {Name = "my_counter", Tags = new MetricTags("test", "my_test")};
+            metrics.Measure.Counter.Increment(counter);
+
+            Thread.Sleep(10000);
+
+            metrics.Measure.Counter.Increment(counter);
+
+            Thread.Sleep(100000); 
+```
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:343]: Please delete the Thread.Sleep lines (they are not relevant), and make the lines to start from same position as var counter line (you can even put a new line after first line and then the 2 metrics.Measure lines one after another)

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 6]
```diff
@@ -0,0 +1,570 @@
...
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:114]: Same thing here :)

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 7]
```diff
@@ -0,0 +1,570 @@
...
+##### Create Scheduler
+
+To create Scheduler, copy and paste the following code into the same function of the code as the MetricsBuilder:
+
+```csharp
+var scheduler = new AppMetricsTaskScheduler(
+                TimeSpan.FromSeconds(15),
+                async () => { await Task.WhenAll(metrics.ReportRunner.RunAllAsync()); });
+            scheduler.Start();
+```
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:90]: last line need to be in same position as first line

File: _source/logzio_collections/_prometheus-sources/dotnet-custom.md [hunk 8]
```diff
@@ -0,0 +1,570 @@
...
```
Comment (ShiranAvidov) [inline _source/logzio_collections/_prometheus-sources/dotnet-custom.md:319]: Same here :)

**Review by ShiranAvidov — CHANGES_REQUESTED** [review]
4 last changes and you are done :)

**Review by ShiranAvidov — APPROVED** [review]

**Review by YajB — APPROVED** [review]
Corrected typos, sentence structure.
Recommendation: Use snippets for blocks of text that appear in multiple locations in documents.

**Pull request merged**